server/application/rest/work_shift.py

```diff
@@ -86,12 +87,32 @@ def work_shifts():
 
         # find workshifts matching the request object
         response = workshift_list_use_case(repo, request_object, user)
-        return Response(
-            json.dumps(response.value, cls=WorkShiftJsonEncoder),
-            mimetype="application/json",
-            status=HTTP_STATUS_CODES_MAPPING[response.response_type],
-        )
-
+        if response.response_type == ResponseTypes.SUCCESS:
+            enriched_shifts = []
+            for work_shift in response.value:
+                # Convert the WorkShift object to JSON
+                work_shift_json = json.loads(json.dumps(work_shift,
+                                                    cls=WorkShiftJsonEncoder))
+                facility_response = get_facility_info_use_case(
+                    work_shift_json["shelter"])
+                if isinstance(facility_response, ResponseSuccess):
```

Comment:
Instead of using isinstance to determine if the response is successful, can you check facility_response.response_type? it's better practice and will be consistent with what we do in other cases?